File: AtlasWeb/Core/Modules/Tibbi_Surec/Kayit_Kabul_Modulu/PatientAdmissionSearch/PatientAdmissionSearchFormViewModel.ts

import { Guid } from "NebulaClient/Mscorlib/Guid";
import { PatientAdmission } from 'NebulaClient/Model/AtlasClientModel';
import { BasePatientSearchResultModel, BasePatientInfoListsModel, BasePatientInfoDet } from "../../PatientSearch/PatientSearchFormViewModel";
import { Type } from 'NebulaClient/ClassTransformer';
import { BaseViewModel } from 'NebulaClient/Model/BaseViewModel';

export class PatientAdmissionSearchFormViewModel extends BaseViewModel {

    public SearchText: string = "";
    @Type(() => PatientInfoListsModel)
    public PatientInfoLists: PatientInfoListsModel;
    @Type(() => PatientSearchResultModel)
    public PatientSearchResult: PatientSearchResultModel;
    @Type(() => Guid)
    public SelectedPatientObjectID: Guid;
    //public SelectedAdmission : PatientAdmissionInfoDet;
    @Type(() => PatientInfoDet)
    public SelectedPatient: PatientInfoDet;


    constructor() {
        super();
        this.PatientSearchResult = new PatientSearchResultModel();
        this.PatientInfoLists = new PatientInfoListsModel();
        //this.SelectedAdmission = new PatientAdmissionInfoDet();
        this.SelectedPatient = new PatientInfoDet();

    }
}

export class PatientSearchResultModel extends BasePatientSearchResultModel {


    public PatientAdmissionInfo: PatientAdmission;

    constructor() {
        super();
    }
}

export class PatientInfoListsModel extends BasePatientInfoListsModel {


    public PatientAdmissionList: PatientAdmissionInfoDet[];
     //public PatientInfo: Patient;
     //public AppointmentInfo: Appointment;
     //public AdmissionSpecOrResOfCurrentDayEpisodes: string = "";
     //public PatientAdmissionInfo: PatientAdmission;

    constructor() {
        super();
        //super.PatientList = new Array<PatientInfoDet>();
        this.PatientAdmissionList = new Array<PatientAdmissionInfoDet>();
    }
}

export class PatientInfoDet extends BasePatientInfoDet
{

}

export class PatientAdmissionInfoDet
{
     public ObjectID: Guid;
     public ActionDate: string;
     public Info: string;
}
// class AppointmentInfo
//{
//    public  ObjectID: Guid;
//}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Kayit_Kabul_Modulu/PatientAdmissionSearch/PatientAdmissionSearchFormViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using TTObjectClasses;

namespace Core.Models
{
    public class PatientAdmissionSearchFormViewModel
    {
        public string SearchText
        {
            get;
            set;
        }

        public PatientSearchResultModel PatientSearchResult
        {
            get;
            set;
        }

        public PatientInfoListsModel PatientInfoLists
        {
            get;
            set;
        }

        //public PatientAdmissionInfoDet SelectedAdmission { get; set; }
        public PatientInfoDet SelectedPatient
        {
            get;
            set;
        }

        public PatientAdmissionSearchFormViewModel()
        {
            this.PatientSearchResult = new PatientSearchResultModel();
            this.PatientInfoLists = new PatientInfoListsModel();
            //this.SelectedAdmission = new PatientAdmissionInfoDet();
            this.SelectedPatient = new PatientInfoDet();
        }
    }

    public class PatientInfoListsModel : BasePatientInfoListsModel
    {
        public List<PatientAdmissionInfoDet> PatientAdmissionList
        {
            get;
            set;
        }

        //public Patient PatientInfo { get; set; }
        //public Appointment AppointmentInfo { get; set; }
        public string AdmissionSpecOrResOfCurrentDayEpisodes
        {
            get;
            set;
        }

        public bool hasGizliHastaKabulRoleID
        {
            get;
            set;
        }

        // public PatientAdmission PatientAdmissionInfo { get; set; }
        public PatientInfoListsModel()
        {
            this.PatientList = new List<PatientInfoDet>();
            this.PatientAdmissionList = new List<PatientAdmissionInfoDet>();
            this.hasGizliHastaKabulRoleID = false;
        //this.PatientInfo = new Patient();
        //this.AppointmentInfo = new Appointment();
        //this.PatientAdmissionInfo = new PatientAdmission();
        }
    }

    public class PatientSearchResultModel : BasePatientSearchResultModel
    {
        public PatientAdmission PatientAdmissionInfo
        {
            get;
            set;
        }
    }

    public class PatientAdmissionInfoDet
    {
        public Guid ObjectID
        {
            get;
            set;
        }

        public string ActionDate
        {
            get;
            set;
        }

        public string Info
        {
            get;
            set;
        }
    }
//public class AppointmentInfo
//{
//    public Guid ObjectID { get; set; }
//}
}